src/helpers.js

```diff
@@ -24,39 +24,66 @@ const promisify = require('@google-cloud/common').util.promisify;
 const gax = require('google-gax');
 
 /*!
- * Find a given image and fire a callback with the appropriate image structure.
+ * Coerce several nicer iterations of "how to specify an image" to the
+ * full sturcture expected by the Vision API.
  *
- * @param {object} image An object representing what is known about the
- *   image.
+ * @param {object|string|Buffer} request An object representing an
+ *   AnnotateImageRequest. May also be a string representing the path (filename
+ *   or URL) to the image, or a buffer representing the image itself.
  * @param {function} callback The callback to run.
  */
-var coerceImage = (image, callback) => {
+let coerceRequest = (request, callback) => {
+  // If request is not an object, then it is a specification of an image
+  // in raw form; coerce it to an object.
+  if (is.string(request) === true) {
+    // Is this a URL or a local file?
+    // Guess based on what the string looks like, and build the full
+    // request object in the correct format.
+    if (request.indexOf('://') === -1 || request.indexOf('file://') === 0) {
+      request = {image: {source: {filename: request}}};
+    }
+    else {
+      request = {image: {source: {imageUri: request}}};
+    }
+  }
+  else if (Buffer.isBuffer(request)) {
+    // Drop the buffer one level lower; it will get dealt with later
+    // in the function. This allows sending <Buffer> and {image: <Buffer>} to
+    // both work identically.
+    request = {image: request};
+  }
+
+  // At this point, request must be an object with an `image` key; if not,
+  // it is an error. If there is no image, throw an exception.
+  if (!is.object(request) || is.undefined(request.image)) {
+    return callback(new Error('No image present.'));
+  }
+
   // If this is a buffer, read it and send the object
   // that the Vision API expects.
-  if (Buffer.isBuffer(image)) {
-    callback(null, {
-      content: image.toString('base64'),
-    });
-    return;
+  if (Buffer.isBuffer(request.image)) {
+    request.image = {content: image.toString('base64')};
   }
 
-  // File exists on disk.
-  if (image.source && image.source.filename) {
-    fs.readFile(image.source.filename, {encoding: 'base64'}, (err, blob) => {
+  // If the file is specified as a filename and exists on disk, read it
+  // and coerce it into the base64 content.
+  if (request.image.source && request.image.source.filename) {
+    fs.readFile(request.image.source.filename, {encoding: 'base64'}, (err, blob) => {
```

Comment:
I think you need to need to remove the `encoding` option from the `readFile` call, otherwise you'll be base64-encoding the content twice.

Reply:
Concur. That must have been broken in 0.12.0 and we never caught it.

Reply:
According to system tests, either one works. I changed it to the default.